go/tools/builders/nogo_main.go

```diff
@@ -279,13 +334,26 @@ func (act *action) execOnce() {
 		factFilter[reflect.TypeOf(f)] = true
 	}
 	pass := &analysis.Pass{
-		Analyzer:          act.a,
-		Fset:              act.pkg.fset,
-		Files:             act.pkg.syntax,
-		Pkg:               act.pkg.types,
-		TypesInfo:         act.pkg.typesInfo,
-		ResultOf:          inputs,
-		Report:            func(d analysis.Diagnostic) { act.diagnostics = append(act.diagnostics, d) },
+		Analyzer:  act.a,
+		Fset:      act.pkg.fset,
+		Files:     act.pkg.syntax,
+		Pkg:       act.pkg.types,
+		TypesInfo: act.pkg.typesInfo,
+		ResultOf:  inputs,
+		Report: func(d analysis.Diagnostic) {
+			pos := act.pkg.fset.Position(d.Pos)
+			for _, rng := range act.nolint {
+				if pos.Filename != rng.from.Filename {
+					continue
+				}
+				if pos.Line < rng.from.Line || pos.Line > rng.to {
+					continue
+				}
+				// Found a nolint range. Ignore the issue.
+				return
+			}
+			act.diagnostics = append(act.diagnostics, d)
+		},
```

Comment:
Nit, now that it's no longer an one-lined, could you please move the func out to above struct?

```
ignoreNoLintReporter := func(...) {
  ...
}
pass := &analysis.Pass{
  ...
  Report: ignoreNoLintReporter,
  ...
}
```